x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/EvaluatorImplementer.java

```diff
@@ -182,8 +183,38 @@ private MethodSpec realEval(boolean blockStyle) {
         {
             processFunction.args.stream().forEach(a -> a.createScratch(builder));
 
+            boolean vectorize = false;
+            String vectorBuilderName = null;
+            if (blockStyle == false && processFunction.warnExceptions.isEmpty() && processOutputsMultivalued == false) {
+                ClassName type = processFunction.resultDataType(false);
+                vectorize = type.simpleName().startsWith("BytesRef") == false;
+                if (vectorize) {
+                    String arrayType = type.simpleName()
+                        .substring(0, type.simpleName().length() - "Vector".length())
+                        .toLowerCase(Locale.ROOT);
+                    builder.addStatement(arrayType + "[] buffer = result.values()");
```

Comment:
I wonder if we could remove `nextIndex` from `Vector.FixedBuilder` and set the `appendFoo` methods take an offset.

Reply:
I was thinking [the same](https://github.com/elastic/elasticsearch/pull/109332#discussion_r1627981386), I'll give it a try